### ISSUE TITLE
RobertaForSequenceClassification doesn't have a "pooler" layer by default in HuggingFace Transformers

### ISSUE THREAD

**Issue by cpcdoy** [guillaume-be/rust-bert#95]
Hi,

While trying to implement a Distilroberta for sequence classification using your library, I noticed that `BertModel::new` creates a `pooler` by default ([link to the code](https://github.com/guillaume-be/rust-bert/blob/master/src/bert/bert_model.rs#L165)) when used with BERT or RoBERTa for sequence classification, while in HuggingFace Transformers they default to no pooling ([link to the code](https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_roberta.py#L954)) which means that your library will be looking for `roberta.pooler.*` weights when there aren't any.

I just wanted to make sure my understanding of what's going on is correct and if there's any way to make this pooler optional.
If not, I can provide a PR to fix this issue.

Thanks for your help

**Comment by cpcdoy** [comment]
I think the following solutions are available to us:

1) Add a boolean parameter `add_pooling_layer` to `BertModel` ([here](https://github.com/guillaume-be/rust-bert/blob/master/src/bert/bert_model.rs#L156)) like they do in HF transformers ([here](https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_bert.py#L726)).
2) Check if `pooler` is present in the `VarStore` and automatically load it or not. This could be dangerous as it might load or not something the user is/isn't expecting, but maybe putting a warning message could make it work (like what is done in HF Transformers when new layers are loaded).
3) Create a new class `RobertaModel` with an `add_pooling_layer ` parameter so that `BertModel `keeps its current constructor.

Personally, I think solution 1 would work best and is the closest to what HF does, but I'd like to have your opinion on this.

**Comment by guillaume-be** [comment]
Hi @cpcdoy ,

Yes - it seems the behaviour of the `BertModel` as implemented in the original library changed. My preferred solution would definitely go in the first direction. I am thinking if there would be a way to make it a non-breaking change.
Instead of updated the existing constructor at https://github.com/guillaume-be/rust-bert/blob/c9e8e960a8d76706a0d0afde5e76d30592b161e8/src/bert/bert_model.rs#L156, an alternative would be to add a constructor like:
```rust
pub fn new_with_optional_pooler<'p, P>(p: P, config: &BertConfig, add_pooling_layer: bool) -> BertModel<T>
where
	P: Borrow<nn::Path<'p>>,
{
	let p = p.borrow();

	let is_decoder = config.is_decoder.unwrap_or(false);
	let embeddings = T::new(p / "embeddings", config);
	let encoder = BertEncoder::new(p / "encoder", config);
        let pooler = if add_pooling_layer {
            Some(BertPooler::new(p / "pooler", config))
        } else {
            None
        }

	BertModel {
		embeddings,
		encoder,
		pooler,
		is_decoder,
	}
}
```

The following would need to be updated:
- BertModel struct field
- RoBERTa models constructors (they seem to never use the pooler)

I am a bit more undecided on the `BertModelOutput`:
https://github.com/guillaume-be/rust-bert/blob/c9e8e960a8d76706a0d0afde5e76d30592b161e8/src/bert/bert_model.rs#L1131

The most idiomatic way would then be to have `pooled_output` be a `Option<Tensor>` instead of a `Tensor`. This could potentially lead to breaking changes if users of the library manipulate the `BertModelOutput` directly, which is rather unlikely (the same way it is unlikely that users would create a `BertModel` directly.

In summary, I would tend to go for an idiomatic refactor of `BertModelOutput` (and therefore also just update the `new` constructor to have optional pooling since this level of the code would have breaking changes anyway).

I could make these updates quickly, unless you started to work on a PR already?

**Comment by cpcdoy** [comment]
So, I actually finished all the changes on my fork, so I'm ready to do a PR and I went in this exact direction you're describing. I'll just add a new constructor like you're saying so that it doesn't break previous changes and it should be good.
